services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -54,6 +55,37 @@ class HakemusService(
         )
     }
 
+    /** Create a johtoselvitys from a hanke that was just created. */
+    @Transactional
```

Comment:
I am not sure, but should this be `@Transactional(propagation = Propagation.MANDATORY)` because we pass an entity as a parameter?

Reply:
I think this would stop us from testing this in a separate integration test, or at least it would be a lot harder.

Reply:
Adding `@Transactional` to test methods or classes means that the tests might not hit the DB at all. Or at least until the test and all its checks are completed. It's possible to start and stop the transaction manually with 

I'm not sure how necessary it is to have an existing transaction when calling the method. I mean, the service tests save the entity without a transaction and then call this method just fine. I tried to make this more explicit by creating a new hanke with just the mandatory fields and definitely without any magic, and the hakemus was created successfully.

```kotlin
val hanke = hankeFactory.saveMinimal(nimi = hakemusNimi)
val otherHanke =
    HankeEntity(id = hanke.id, nimi = hanke.nimi, hankeTunnus = hanke.hankeTunnus)

hakemusService.createJohtoselvitys(otherHanke, USERNAME)

assertThat(applicationRepository.findAll()).single().all {
    ...
}
```
I think if we follow the basic guideline of "Don't return entities from a transaction", this should be completely safe. This guideline is one of the reasons `createJohtoselvitys` returns a Hakemus, and not the ApplicationEntity it created.

Reply:
Ok, thank you for the clarification.